checks.d/network.py

```diff
@@ -440,3 +446,117 @@ def _parse_solaris_netstat(self, netstat_output):
+        proc_name = '/proc/sys/net/ipv4/ip_conntrack_max'
+        try:
+            proc = open(proc_name,'r')
+            try:
+                lines = proc.readline()
+            finally:
+                proc.close()
+            return int(lines)
+        except IOError:
+            return 0
+
+    @staticmethod
+    def _linux_sockstat_get_conntrack_count():
+        proc_name = '/proc/sys/net/ipv4/netfilter/ip_conntrack_count'
+        try:
+            proc = open(proc_name,'r')
+            try:
+                lines = proc.readline()
+            finally:
+                proc.close()
+            return int(lines)
+        except IOError:
+            return 0
+
+    @staticmethod
+    def _linux_sockstat_get_max_orphans(self):
+        proc_name = '/proc/sys/net/ipv4/tcp_max_orphans'
+        try:
+            proc = open(proc_name,'r')
+            try:
+                lines = proc.readline()
+            finally:
+                proc.close()
+            return int(lines)
+        except IOError:
+            self.log.debug("Unable to read %s." % proc_name)
+
+    @staticmethod
+    def _linux_sockstat_get_size_of_page():
+        return long(check_output(["/usr/bin/getconf", "PAGESIZE"]))
+
+    @staticmethod
+    def _linux_sockstat_get_max_memory(self):
+        sizeOfPageInBytes = Network._linux_sockstat_get_size_of_page()
+        tcpMemFilePath = '/proc/sys/net/ipv4/tcp_mem'
+        try:
+            proc = open(tcpMemFilePath,'r')
+            try:
+                tcpmem = proc.readlines()
+            finally:
+                proc.close()
+
+        except IOError:
+            self.log.debug("Unable to read %s." % tcpMemFilePath)
+        tcpmemline = tcpmem[0].split()
+        maxMemoryInBytes = sizeOfPageInBytes * tcpmemline[2]
+        return long(maxMemoryInBytes)
+
+    @staticmethod
+    def _linux_sockstat(self):
+        proc_name = '/proc/net/sockstat'
+        try:
+            proc = open(proc_name,'r')
+            try:
+                lines = proc.read()
+            finally:
+                proc.close()
+            reg = re.compile("sockets: used (?P<sockets_used>\d+)\n"
+                        "TCP: inuse (?P<tcp_inuse>\d+) orphan (?P<orphans>\d+)"
+                        " tw (?P<tw_count>\d+) alloc (?P<tcp_sockets>\d+)"
+                        " mem (?P<tcp_pages>\d+)\n"
+                        "UDP: inuse (?P<udp_inuse>\d+)"
+                        "(?: mem (?P<udp_pages>\d+))?\n"
+                        "(?:UDPLITE: inuse (?P<udplite_inuse>\d+)\n)?"
+                        "RAW: inuse (?P<raw_inuse>\d+)\n"
+                        "FRAG: inuse (?P<ip_frag_nqueues>\d+)"
+                        " memory (?P<ip_frag_mem>\d+)\n")
+            match = re.match(reg, lines)
+            if not match:
+                self.log.debug("Unable to parse %s." % proc_name)
+                return
+
+            values = []
+            values.append(match.group("sockets_used"))
```

Comment:
Unnecessary to use append so many times.

Instead try something like:
```
metrics = dict()
with group, metric in LINUX_SOCKSTAT_METRICS:
    metrics[metric] = match.group(group)
```

Assuming of course you've already followed the next comment about defining a `LINUX_SOCKSTAT_METRICS` variable for metric names and regex groups. See comment below